src/worker/functions/commit-files.js

```diff
@@ -0,0 +1,109 @@
+const getCurrentCommit = async (octokit, username, repo, branch = 'main') => {
+  const { data: refData } = await octokit.request(
+    `GET /repos/${username}/${repo}/git/ref/heads/${branch}`,
+    {
+      owner: username,
+      repo: repo,
+      ref: `heads/${branch}`,
+    }
+  );
+
+  const commitSha = refData.object.sha;
+  const { data: commitData } = await octokit.request(
+    `GET /repos/${username}/${repo}/git/commits/${commitSha}`,
+    {
+      owner: username,
+      repo: repo,
+      commit_sha: commitSha,
+    }
+  );
+  return {
+    commitSha,
+    treeSha: commitData.tree.sha,
+  };
+};
+
+const createBlobForFile = async (octokit, username, repoName, files) => {
+  let blobDataPromises = [];
+  for (const file of files) {
+    let reader = new FileReader();
+    await reader.readAsArrayBuffer(file);
+    const promise = new Promise((resolve, reject) => {
+      reader.onload = async () => {
+        const imageBuffer = reader.result;
+        const byteArray = new Uint8Array(imageBuffer);
+        const blob = new Blob([byteArray]);
```

Comment:
Since you know the file types we allow (i.e., `image/*`), why don't we pass the `mimeType` here too as second arg: https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob

Reply:
Will be addressed accordingly